.github/workflows/ci.yaml

```diff
@@ -43,9 +43,9 @@ jobs:
       matrix:
         python-version: ["3.9"]
     steps:
-    - uses: actions/checkout@v4
+    - uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # 4.2.2
```

Comment:
anything starts with `action/` is considered safe as it originates from github themselves. therefor no need to pin to hash. can be left with tag. that would save us troubles updating steps